src/vsg/io/FileSystem.cpp

```diff
@@ -278,3 +279,47 @@ bool vsg::makeDirectory(const Path& path)
 
     return true;
 }
+
+Path vsg::executableFilePath(void)
+{
+    Path path;
+    char buf[1024];
+
+#if defined(WIN32)
+    DWORD result = GetModuleFileName(NULL, buf, sizeof(buf)-1);
+    if (result && result < sizeof(buf))
+        path = buf;
+#else
+    ssize_t len = readlink("/proc/self/exe", buf, sizeof(buf)-1);
```

Comment:
Will this work under Android , macOS,and iOS?

Reply:
I added related cpp macros as place holder for currently unimplemented platforms. For mac osx I did found a hint how to implement that.